Header header
bool TD_signal
bool TU_signal